packages/material-ui/src/Typography/Typography.d.ts

```diff
@@ -13,6 +13,7 @@ export interface TypographyProps
   headlineMapping?: { [type in Style]: string };
   noWrap?: boolean;
   paragraph?: boolean;
+  useNextVariants?: boolean;
```

Comment:
I think that we should only have this option in the theme. What's the use case for having it in the property? Progressive migration? I'm fine dropping that for the simplicity we get in exchange.

Reply:
It's probably a good idea. Reduces added code (and potential risk for bugs). I wouldn't recommend mixing different typography in the first place. It's hopefully a niche use case and can always be solved by using different themes.

Reply:
My initial use case was for the `Typography` test suite were I didn't want to wrap every `Typography` with a `MuiThemeProvider` so that it doesn't throw deprecation warnings. A few iterations later this can be achieved by just passing a theme with `ignoreDeprecationWarnings`. It's a little hacky but given that this is only existing in our codebase for a limited period of time I'm ok with a brittle test suite.